src/main/java/org/elasticsearch/search/aggregations/bucket/children/ParentToChildrenAggregator.java

```diff
@@ -66,7 +66,8 @@
     private final LongObjectPagedHashMap<long[]> parentOrdToOtherBuckets;
     private boolean multipleBucketsPerParentOrd = false;
 
-    private List<AtomicReaderContext> replay = new ArrayList<>();
+    // This needs to be a Set to avoid duplicate reader context entries (#setNextReader(...) can get invoked multiple times with the same reader context)
+    private Set<AtomicReaderContext> replay = new HashSet<>();
```

Comment:
I think it should be a LinkedHashSet since we rely on doc ID order for tie-breaking.